.github/workflows/dispatch-matrix-tests.yaml

```diff
@@ -46,7 +46,7 @@ jobs:
       - name: Set up python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.10'
+          python-version: '3.12'
```

Comment:
I'd add a comment, maybe a link to a github issue where Python compatibility is tracked.